### PR TITLE
Do not convert boolean values to string in deep_string_coerce function

### LINKED ISSUE #25232: enable_elastic_disk property incorrectly mapped when making a request to Databricks
### Apache Airflow version

2.2.2

### What happened

When using `apache-airflow-providers-databricks` in version 2.2.0 I am sending a request to databricks to submit a job.
https://docs.databricks.com/dev-tools/api/latest/jobs.html#operation/JobsCreate -> `api/2.0/jobs/runs/submit`

Databricks is expecting a boolean on a property `enable_elastic_disk` while `airflow-databricks-provider` sends a string.

```
new_cluster = {
    "autoscale": {"min_workers": 2, "max_workers": 5},
    "spark_version": "10.4.x-scala2.12",
    "aws_attributes": {
        "first_on_demand": 1,
        "availability": "SPOT_WITH_FALLBACK",
        "zone_id": "auto",
        "spot_bid_price_percent": 100,
    },
    "enable_elastic_disk": True, 
    "driver_node_type_id": "r5a.large",
    "node_type_id": "c5a.xlarge",
    "cluster_source": "JOB",
}
```

And the property `enable_elastic_disk` is not set on databricks side. I did also the same request to databricks from a Postman and the property was set to `true` which means that the problem does not lie on databricks side.
```
{
    "name": "test",
    "tasks": [
        {
            "task_key": "test-task-key",
            "notebook_task": {
                "notebook_path": "path_to_notebook"
            },
            "new_cluster": {
                "autoscale": {"min_workers": 1, "max_workers": 2},
                "cluster_name": "",
                "spark_version": "10.4.x-scala2.12",    
                "aws_attributes": {
                    "first_on_demand": 1,
                    "availability": "SPOT_WITH_FALLBACK",
                    "zone_id": "auto",        
                    "spot_bid_price_percent": 100                    
                },
                "driver_node_type_id": "r5a.large",
                "node_type_id": "c5a.xlarge",
                "enable_elastic_disk": true,                     
                "cluster_source": "JOB"                
            }
        }
    ]    
}
```

I have tried to find the problem and it apparently is this line. Before executing the line `enable_elastic_disk` is `True` of type boolean but after it becomes a string `'True'` which databricks does not parse.
https://github.com/apache/airflow/blob/1cb16d5588306fcb7177486dc60c1974ea3034d4/airflow/providers/databricks/operators/databricks.py#L381

### What you think should happen instead

After setting property `enable_elastic_disk` it should be propagated into databricks but it's not.

### How to reproduce

Try to run:

```
new_cluster = {
    "autoscale": {"min_workers": 2, "max_workers": 5},
    "spark_version": "10.4.x-scala2.12",
    "aws_attributes": {
        "first_on_demand": 1,
        "availability": "SPOT_WITH_FALLBACK",
        "zone_id": "auto",
        "spot_bid_price_percent": 100,
    },
    "enable_elastic_disk": True, 
    "driver_node_type_id": "r5a.large",
    "node_type_id": "c5a.xlarge",
    "cluster_source": "JOB",
}

notebook_task = {
    "notebook_path": f"/Repos/path_to_notebook"/main_asset_information",
    "base_parameters": {"env": env},
}


asset_information = DatabricksSubmitRunOperator(
        task_id="task_id"
        databricks_conn_id="databricks",
        new_cluster=new_cluster,
        notebook_task=notebook_task,
    )
```

Make sure airflow connection named `databricks` is set and check whether databricks has the property set.

After executing there is a need to check whether the property is set on databricks we can do it by using endpoint:
`https://DATABRICKS_HOST/api/2.1/jobs/runs/get?run_id=123`

### Operating System

MWWA

### Versions of Apache Airflow Providers

`apache-airflow-providers-databricks` in version 2.2.0

### Deployment

MWAA

### Deployment details

_No response_

### Anything else

That's a permanent and repeatable problem. It would be great if this fix could be attached to lower versions for example `2.2.1`, because I am not sure when AWS decides to upgrade to the latest airflow code and I am also not sure if installing higher versions of databricks provider on airflow `2.2.2` will not cause issues.

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by jgr-trackunit**
When we send request to Databricks with quoted bool values like: `"False"` and `"True"` Databricks interprets them as a strings which is undesired behaviour. In that case we need to send requests with real boolean values.

closes: [#25232](https://github.com/apache/airflow/issues/25232)
related:[#25232](https://github.com/apache/airflow/issues/25232)

- Unittests have been updated, unit tests have been run with `Passed` result
- Docstring for `deep_string_coerce` function has been updated
- Command `breeze static-checks --all` passed

@potiuk @aru-trackunit

### CODE REVIEW

**Review by potiuk — APPROVED** [review]
LGTM. @alexott  WDYT?

**Review by alexott — APPROVED** [review]
Patch looks good to me. I always had a problem with that function because it’s more a workaround around the render_template function. If render_template can handle non-string types gracefully, then we don’t need this function at all.

p.s. in my other projects I’ve implemented templates, but it was other way around, string templates were converted into non-string types if variables were of these types

**Comment by uranusjr** [comment]
The logic is correct, but the function should not be named `deep_string_coerce`.

**Comment by jgr-trackunit** [comment]
HI @potiuk @uranusjr 

So shall I rename that function? Should it be the part of another PR? Can I merge it?

**Comment by potiuk** [comment]
> HI @potiuk @uranusjr
> So shall I rename that function? Should it be the part of another PR? Can I merge it?

Yes. rename it please. Why do you ask?  Is there any doubt about it ? I thinkt it was clear and justified ask.

**Comment by jgr-trackunit** [comment]
> > HI @potiuk @uranusjr
> > So shall I rename that function? Should it be the part of another PR? Can I merge it?
> 
> Yes. rename it please. Why do you ask? Is there any doubt about it ? I thinkt it was clear and justified ask.

I was just not sure if we want to change that function name in this PR. :)

I would like to propose new name: either `normalise_json_content` or `normalise_json_content`, what do you think @potiuk @uranusjr? Would be that name consistent enough?

**Comment by potiuk** [comment]
> > > HI @potiuk @uranusjr
> > > So shall I rename that function? Should it be the part of another PR? Can I merge it?
> > 
> > 
> > Yes. rename it please. Why do you ask? Is there any doubt about it ? I thinkt it was clear and justified ask.
> 
> I was just not sure if we want to change that function name in this PR. :)
> 
> I would like to propose new name: either `normalise_json_content` or `normalise_content`, what do you think @potiuk @uranusjr? Would be that name consistent enough?

both are better :). Pick one you like :D

File: airflow/providers/databricks/utils/databricks.py [hunk 1]
```diff
@@ -23,16 +23,25 @@
...
-    coerce = deep_string_coerce
-    if isinstance(content, str):
+    normalise = normalise_json_content
+    if isinstance(
+        content,
+        (
+            str,
+            bool,
+        ),
+    ):
```
Comment (uranusjr) [inline airflow/providers/databricks/utils/databricks.py]:
```suggestion
    if isinstance(content, (str, bool)):
```

**Review by uranusjr — APPROVED** [review]

**Comment by jgr-trackunit** [comment]
Who can merge it now? What about the release process? I'm interested to have this changes released ASAP, so I can help somehow with that, ie. testing it in real env. Could you tell me who is the release manager?

**Comment by potiuk** [comment]
> Who can merge it now? What about the release process? I'm interested to have this changes released ASAP, so I can help somehow with that, ie. testing it in real env. Could you tell me who is the release manager?

It's quite rude way of asking for things. I would expect you politely ask rather than demand things. 

You have to be patient (and also empathetic to the process we have and others). This is a very basic thing when you interface with open-source software that you should learn. 

We release providers roughly once a month. This is clearly described in the docuementation. There is no way this can be sped up because "you" need it. If there is big problem for the community impacting a lot of people, we might do ad-hoc releases but this is not at all needed here.

The release process is described here: https://github.com/apache/airflow/blob/main/README.md#release-process-for-providers

And if you REALLY want to do things ASAP there is no problem for you to build and use the provider with your changes built yourself. Building provider packages is easy and `breeze` tool helps with that: https://github.com/apache/airflow/blob/main/BREEZE.rst#preparing-provider-packages

You have to remember that there is nothing like ASAP in open-source world. You (and your company) paid exactly 0 USD for the free software (that many people deliver for free). So your demands have exactly 0 worth here. 

BTW. I am the release manager of providers.

**Comment by potiuk** [comment]
And yes. We will ask you to test it when we release it together with other providers. This is also part of the standard process - you can see how it was before: https://github.com/apache/airflow/issues?q=is%3Aissue+status+of+testing+is%3Aclosed+label%3A%22testing+status%22

**Comment by jgr-trackunit** [comment]
Really sorry for my previous message I didn't want to offend yourself and I didn't demand anything. As well I'm not trying to force you to release it now.

So apologies one more time for that, It was not the goal of that message. 

I see that chapter is missing: https://github.com/apache/airflow/blob/main/README.md#release-process-for-providers (but I read it ealier)

Thanks for the 2nd link.

**Comment by potiuk** [comment]
> So apologies one more time for that, It was not the goal of that message.

No worries. Apologies for harsh tone. It's a bit of being sensitive after some "real" demands I saw before. Thanks for being sensitive and receptive to it.

> I see that chapter is missing: https://github.com/apache/airflow/blob/main/README.md#release-process-for-providers (but I read it ealier)

It's there, just GitHub having problems with scrolling down to it's own anchors (Sometimes) - seen that many times. Usually reloading it the second time works :)

**Comment by uranusjr** [comment]
```
tests/providers/databricks/operators/test_databricks.py::TestDatabricksRunNowOperator::test_exec_failure_with_message:
AttributeError: module 'airflow.providers.databricks.utils.databricks' has no attribute 'deep_string_coerce'
```

**Comment by jgr-trackunit** [comment]
> tests/providers/databricks/operators/test_databricks.py::TestDatabricksRunNowOperator::test_exec_failure_with_message:
> AttributeError: module 'airflow.providers.databricks.utils.databricks' has no attribute 'deep_string_coerce'

Fixed

**Comment by uranusjr** [comment]
The spell checker is American

```
 File path: apache-airflow-providers-databricks/_api/airflow/providers/databricks/utils/databricks/index.rst
Incorrect Spelling: 'Normalise'
Line with Error: 'Normalise content or all values of content if it is a dict to a string. The'
Line Number: 23
```

**Pull request merged**